### ISSUE TITLE
Dynamic import with extra parenthesis not captured

### ISSUE THREAD

**Issue by privatenumber** [guybedford/es-module-lexer#156]
When the dynamic import argument is wrapped in a parenthesis, the statement range is not correctly captured:

```js
import { init, parse } from 'es-module-lexer';

await init;

const source = 'import(("asdf"))';

const [imports] = parse(source, 'optional-sourcename');
const [imp] = imports;
const dynamicImportStatement = source.slice(imp.ss, imp.se);
// import(("asdf")
// Expected to be: import(("asdf"))

if (dynamicImportStatement !== source) {
  console.log('Full statement was not captured:', dynamicImportStatement);
}
```

Reproduction:
https://stackblitz.com/edit/stackblitz-starters-zzxwkp?file=index.mjs

**Comment by guybedford** [comment]
Hey @privatenumber, yes this is by design - we don't parse arbitrary expressions, so from the perspective of the lexer, `('string')` is an expression, while `'string'` is a direct static string.

In the dynamic expression cases, you always do just need to apply a custom parser, but at least there is an expression range you can apply it to.

**Comment by privatenumber** [comment]
The problem is that the expression range is incorrect, which results in invalid JS.

e.g. a parser cannot correctly parse `import(("asdf")` because it's missing the closing parenthesis `)`.

**Comment by guybedford** [comment]
Okay, right, if it’s the wrong offset for a slice function then that’s a
buf we should fix.

**Comment by guybedford** [comment]
Released in 1.3.1.

**Comment by privatenumber** [comment]
Thank you @guybedford !
